apps/spruce/src/pages/waterfall/BuildRow.tsx

```diff
@@ -75,7 +94,14 @@ export const BuildRow: React.FC<Props> = ({
           {displayName}
         </StyledLink>
       </BuildVariantTitle>
-      <BuildGroup data-cy="build-group">
+      <VisibilityContainer
```

Comment:
This is awesome, I think it also really  improves the speed of the initial page render 👏👏